### PR TITLE
Added webcrypto template

### PR DESCRIPTION
**Pull request by nataliescottdavidson**

### CODE REVIEW

File: templates/typescript/webcrypto.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+let subtle = self.crypto.subtle;
```
Comment (ispivey) [inline templates/typescript/webcrypto.ts]: In most of the code below, you call `self.crypto.subtle` directly rather than using this shortcut. That seems fine, and it doesn't make the code hard to follow! I'd suggest getting rid of this variable declaration, and changing its usages below (on lines 29 and 47) to reference `self.crypto.subtle` directly. In this kind of example content, I think being clear and using the underlying API as directly as possible is best.

File: templates/javascript/webcrypto.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+"use strict";
```
Comment (ispivey) [inline templates/javascript/webcrypto.js:1]: I wonder why this ended up in the transpiled JavaScript... `tsc` should only prepend `"use strict"` for non-ES6 targets.

From the project root, can you try running
```bash
npm i && npm run transpile && npm run lint
```
and seeing if you get a JS output without `"use strict"` prepended?

**Review by ispivey — CHANGES_REQUESTED** [review]
This looks great – couple small things! Getting the transpiling/linting working right is the big one. The transpile/lint process is pretty new, so if you try it again and it doesn't resolve the issue, please just let me know and I'll debug further.

File: templates/typescript/webcrypto.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  return new Response()
+}
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest())
+})
+
+async function generateSignandVerify(algorithm: any) {
+  let rawMessage = "alice and bob"
+  let key = await subtle.generateKey(algorithm, true, ["sign", "verify"]);
```
Comment (kristianfreeman) [inline templates/typescript/webcrypto.ts]: could we cast this `as CryptoKey` here so we don't have to do it in the `subtle` calls?

File: templates/typescript/webcrypto.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    algorithm,
+    key as CryptoKey,
+    signature,
+    encoded
+  );
+  return result;
+}
+
+async function generateEncryptDecrypt(algorithm: any, msg: string) {
+  let key = await subtle.generateKey(
```
Comment (kristianfreeman) [inline templates/typescript/webcrypto.ts]: same question as above

Comment (nataliescottdavidson) [reply templates/typescript/webcrypto.ts]: Done.

File: templates/typescript/webcrypto.ts (hunk 1 above)
Comment (nataliescottdavidson) [reply templates/typescript/webcrypto.ts]: Done.

File: templates/javascript/webcrypto.js (hunk 2 above)
Comment (nataliescottdavidson) [reply templates/javascript/webcrypto.js:1]: I still got "use strict"

Comment (nataliescottdavidson) [reply templates/javascript/webcrypto.js:1]: I've just removed this manually. I deleted existing js files and re ran "npm transpile" to see if my environment was prepending this to all generated js files. Nope, just this one. Also checked my webcrypto.toml file to see if there was anything funky- it matches the other metadata files pretty closely.

**Comment by ispivey** [comment]
Discussed separately, and we're going to leave `'use strict'` in there for the moment until I or someone else has a bit of time to figure out why that's there. Otherwise lgtm, and with that change we'll merge.

**Review by ispivey — APPROVED** [review]
